demo/api/src/app.module.ts

```diff
@@ -143,6 +145,24 @@ export class AppModule {
                     directory: `${config.blob.storageDirectoryPrefix}-public-uploads`,
                     acceptedMimeTypes: ["application/pdf", "application/x-zip-compressed", "application/zip"],
                 }),
+                ...(config.contentGeneration.apiKey && config.contentGeneration.url && config.contentGeneration.deploymentId
+                    ? [
+                          ContentGenerationModule.register(
+                              new OpenAiContentGenerationService({
+                                  generateAltText: {
+                                      apiUrl: config.contentGeneration.url,
+                                      apiKey: config.contentGeneration.apiKey,
+                                      deploymentId: config.contentGeneration.deploymentId,
+                                  },
+                                  generateImageTitle: {
+                                      apiUrl: config.contentGeneration.url,
+                                      apiKey: config.contentGeneration.apiKey,
+                                      deploymentId: config.contentGeneration.deploymentId,
+                                  },
+                              }),
```

Comment:
You can't do that. You are working around Nest's Dependency Injection. 

You must either provide the options via DI or implement a factory for OpenAiContentGenerationService. @johnnyomair What do you think is better?

Reply:
As discussed, there should always be a custom ContentGenerationService in the application. This service can then use the OpenAiContentGenerationService provided by the library. Something like:

```ts
class ContentGenerationService implements ContentGenerationServiceInterface {
  constructor(
    private readonly openAiContentGenerationService: OpenAiContentGenerationService,
    @Inject(CONFIG) private readonly config: Config
  ) {
    this.openAiContentGenerationService.init({
      apiUrl: config.contentGeneration.url,
      apiKey: config.contentGeneration.apiKey,
      deploymentId: config.contentGeneration.deploymentId,
    });
  }

  async generateAltText(args) {
    return this.openAiContentGenerationService.generateAltText(args);
  }
}
```

Using inheritance would also be possible:

```ts
class ContentGenerationService implements OpenAiContentGenerationService {
  constructor(
    @Inject(CONFIG) private readonly config: Config
  ) {
    super({
      apiUrl: config.contentGeneration.url,
      apiKey: config.contentGeneration.apiKey,
      deploymentId: config.contentGeneration.deploymentId,
    });
  }
}
```

but that would eliminate the possibility of mixing different content generation services (OpenAi, Llama etc.)

Reply:
@johnnyomair I implemented another solution for the `ContentGenerationModule` here: https://github.com/vivid-planet/comet/pull/1896 

Is that what you had in mind?

Reply:
Yes, it's better. However, I'd still change some details. See my review of #1896.